sentry/src/main/java/io/sentry/TransactionContexts.java

```diff
@@ -1 +1,93 @@
+package io.sentry;
 
+import io.sentry.protocol.App;
+import io.sentry.protocol.Browser;
+import io.sentry.protocol.Device;
+import io.sentry.protocol.Gpu;
+import io.sentry.protocol.OperatingSystem;
+import io.sentry.protocol.SentryRuntime;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public final class TransactionContexts {
```

Comment:
while I agree that this would avoid all sort of NPE around`Contexts` and `trace`, looks like a junk of code duplication, wondering if just moving the `Contexts` inheritance away and adding an `other` field in there would not solve the problem for good (not adding a new class `TransactionContexts`).
Also, `Contexts` would need to become @NotNull by default on SentryBaseEvent.

maybe @bruno-garcia would like to chime in

Reply:
Then we would eliminate in practical terms the possibility of NPE but we won't get away with warnings in the IDE.